xtrabackup_snap_path.patch

```diff
@@ -0,0 +1,38 @@
+--- a/storage/innobase/xtrabackup/src/xtrabackup.cc
++++ b/storage/innobase/xtrabackup/src/xtrabackup.cc
+@@ -4196,6 +4196,19 @@ static void cleanup_mysql_environment() {
+   mysql_mutex_destroy(&LOCK_replica_list);
+ }
+ 
++void prepend_snap_path(char *path) {
```

Comment:
@jnsgruk not sure if this is the kind of workaround you thought about!

Reply:
Ah interesting - that's certainly one way of doing it! I was thinking a more superficial patch just in the snap config ,but this might turn out to be more useful.

NB. we should probably be careful introducing functions like `strcpy` which is generally considered pretty unsafe

Reply:
re: strcpy. Maybe use memcpy() ?

Reply:
There are historical holly wars on "strcpy VS strncpy", e.g. http://the-flat-trantor-society.blogspot.com/2012/03/no-strncpy-is-not-safer-strcpy.html